jupyter_elfinder/elfinder.py

```diff
@@ -391,28 +360,109 @@ def run(
 
         return self.http_status_code, self.http_header, self.http_response
 
+    def __places(self) -> None:
+        if "targets[]" not in self._request:
+            self._response["error"] = "Invalid parameters"
+            return
+
+        targets = self._request["targets[]"]
+        files = []
+        for target in targets:
+            path = self.__find(target)
+            files.append(self.__info(path))
+        self._response["files"] = files
+
     def __open(self) -> None:
         """Open file or directory."""
-        # try to open file
-        if "tree" not in self._request and "current" in self._request:
-            cur_dir = self.__find_dir(self._request["current"], None)
+        # try dir
 
-            if not cur_dir:
-                self.http_status_code = 404
-                self.http_header["Content-type"] = "text/html"
-                self.http_response = "File not found"
+        # initialized = len(self._cached_path) > 0
+        if "init" in self._request and self._request["init"]:
```

Comment:
Invert this check. Always prefer guard clause when possible. It saves on indentation.
```py
if not self._request.get("info") and not self._request.get("target"):
    self._response["error"] = "Invalid parameters"
    return
...
```